harness/tests/experiment/fixtures/pytorch_onevar_model.py

```diff
@@ -92,6 +97,8 @@ def triangle_label_sum(updates: List) -> Any:
 
 
 class OneVarTrial(pytorch.PyTorchTrial):
+    _dataset_cls: Type[torch.utils.data.Dataset] = OnesDataset
+    _dataset_args = ()
```

Comment:
Let's avoid this sort of magic for the dataset.  The reason we should avoid it is because magic generally makes things harder to reason about.

Full disclosure, it was I who added the `_searcher_metric`.  It is intentionally breaking the Trial abstraction in a way that makes writing unit tests a lot easier, because sometimes the unit testing framework needs to know this information as part of testing.  It's a hack, but it's a useful hack.

The difference is that nobody outside of the Trial class needs to know what the `_dataset_cls` is.  It adds an extra layer of indirection without any real gain.

Why not just subclass the `build_training_data_loader()` instead?

Reply:
I'm fine with what you suggest if I can ignore the code block inside `elif self.hparams["dataloader_type"] == "torch":`. If I can't, that's a lot of repeated code that I'd be sad about. ;'(

Reply:
Let me flip the question on you: can you find what code would use that, and tell me if it's safe to ignore?

Reply:
Methinks so: the custom (i.e. non-Determined) dataloader is only used in `test_require_disable_dataset_reproducibility` and `test_custom_dataloader`.
I'm gonna go ahead with your suggestion.